frontend/public/components/routes.tsx

```diff
@@ -85,12 +85,89 @@ export const RouteHostname: React.SFC<RouteHostnameProps> = ({obj}) => <div>
   }
 </div>;
 
+export const RouteStatus: React.SFC<RouteHostnameProps> = ({obj: route}) => {
+  let admittedCondition: boolean = false;
+
+  const statusIconStyle = {
+    paddingRight: '6px',
+    width: '16px'
+  };
+
+  if (!route.status || !route.status.ingress) {
+    return <span>
+      <div className="fa fa-hourglass-half" style={statusIconStyle} aria-hidden="true"></div>
+      Pending
+    </span>;
+  }
+
+  _.each(route.status.ingress, (ingress) => {
+    let condition: any = _.find(ingress.conditions, {type: 'Admitted', status: 'True'});
+    if (condition) {
+      admittedCondition = true;
+    }
+  });
+
+  return <React.Fragment>
+    {admittedCondition && <span>
+      <div className="fa fa-check text-success" style={statusIconStyle} aria-hidden="true"></div>
+      Accepted
+    </span>}
+    {!admittedCondition && <span>
+      <div className="fa fa-times text-danger" style={statusIconStyle} aria-hidden="true"></div>
+      Rejected
+    </span>}
+  </React.Fragment>;
+};
+
+const addTLSWarnings = (route: any, warnings: string[]) => {
+  if (!route.spec || !route.spec.tls) {
+    return;
+  }
+
+  if (!route.spec.tls.termination) {
+    warnings.push('Route has a TLS configuration, but no TLS termination type is specified. TLS will not be enabled until a termination type is set.');
+  }
+
+  if (route.spec.tls.termination === 'passthrough' && route.spec.path) {
+    warnings.push(`Route path "${ route.spec.path }" will be ignored since the route uses passthrough termination.`);
+  }
+};
+
+const addIngressWarnings = (route: any, warnings: string[]) => {
+  if (!route.status) {
+    return;
+  }
+
+  _.each(route.status.ingress, (ingress) => {
+    let condition: any = _.find(ingress.conditions, { type: 'Admitted', status: 'False' });
+    if (condition) {
+      let message = `Requested host '${ ingress.host || '<unknown host>' }' was rejected by the router.`;
+      if (condition.message || condition.reason) {
+        message += ` Reason: ${ condition.message || condition.reason }.`;
+      }
+      warnings.push(message);
+    }
+  });
+};
+
+export const RouteWarnings: React.SFC<RouteHostnameProps> = ({obj: route}) => {
+  let warnings: string[] = [];
+
+  if (!route) {
+    return;
+  }
+
+  addTLSWarnings(route, warnings);
+  addIngressWarnings(route, warnings);
+
+  return <div style={{whiteSpace: 'pre-wrap'}}>{warnings.join('\n')}</div>;
```

Comment:
Don't use inline style.

I'd suggest something like:

```jsx
{warnings.map(warning => <div>{warning}</div>)}
```

Reply:
The ingress warning may be a string with \n's, thus the `pre-wrap`.

Reply:
You can still use pre-wrap on the divs. It shouldn't be an inline style though.